lib/queue.rb

```diff
@@ -1,28 +1,52 @@
 class Queue
 
-  def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+  def initialize(size = 10)
+    @store = Array.new(size)
+    @front = 0
+    @back = 0
+
+    @is_empty = true
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @front == @back && !@is_empty
+      raise ArgumentError, "Queue is full"
+    end
+
+    @is_empty = false if @is_empty
+    @store[@back] = element
+    @back = (@back + 1) % @store.length
   end
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    removed_value = @store[@front]
+    @store[@front] = nil
+    @front = (@front + 1) % @store.length
+
+    @is_empty = true if @front == @back 
+
+    return removed_value
   end
 
   def front
-    raise NotImplementedError, "Not yet implemented"
+    return @front
   end
 
   def size
```

Comment:
This works, but I do want to present you with a question. 

Why make is_empty a boolean and instead make it a count, when you add one it goes up and when you dequeue you reduce it by one.  That would reduce the time complexity of size and is_empty? could just be checking if the value is 0.